vendor/strega/smart-valve-codec.yaml

```diff
@@ -0,0 +1,52 @@
+# Uplink decoder decodes binary data uplink into a JSON object (optional)
+# For documentation on writing encoders and decoders, see: https://thethingsstack.io/integrations/payload-formatters/javascript/
+uplinkDecoder:
+  fileName: smart-valve.js
+  # Examples (optional)
```

Comment:
Will you provide examples? Do you need help with that?

Reply:
When i give the example the code is giving an error.

My input is as follows: "3330313131" and port is 4

The output should be 
```
{
  "Battery": 63,
  "Cable": "0",
  "Class": "0",
  "DI_0": "0",
  "DI_1": "0",
  "Fraud": "0",
  "Leakage": "0",
  "Port": 4,
  "Power": "0",
  "Process": "true",
  "Status": "00000001",
  "Tamper": "0",
  "Valve": "1"
}
```

when i check direct in TTN or TTI i am getting the exact output, but not here.

Can you help me how to make it?

Reply:
You need to pass the input bytes as array. If the input you gave is hex, you should pass it like `[0x33, 0x30, 0x31, 0x31, 0x31]`

Reply:
Can you please review the latest file. and approve it. We have not added example to it. If required we will add it later.

Reply:
Dear Johan. Any updates for the latest and corrected file?

Reply:
Hi, I review files and they look fine, we only need last aproval, please don´t forget to add the example and if you need any support on this please tell me.